astropy/io/misc/tests/test_hdf5.py

```diff
@@ -566,6 +566,14 @@ def test_read_h5py_objects(tmpdir):
 
     f.close()         # don't raise an error in 'test --open-files'
 
+@pytest.mark.skipif('not HAS_H5PY')
+def test_write_to_Hdf5(tmpdir):
+    test_file = str(tmpdir.join('test.hdf5'))
+    t1 = Table()
+    t1['x'],t1['y'] = ['1', '2', '3'],['a', 'b', 'c']
+    t1.write(test_file, path='the_table',overwrite=True)
+    assert np.all(t1['x']==['1','2','3'])
+    assert np.all(t1['y']==['a','b','c'])
```

Comment:
You probably need to read the written file to check that it was correctly encoded.

Reply:
As @saimn says, this is not testing the correct output to hdf5, but the table you just created!

Reply:
Note that it is also interesting to check that the input file was not modified.

Reply:
Yes I should have read the written file , i will make the changes. I actually thought that the issue was storing the table(having native py3 strings) in hdf5 so i just showed that after this fix no exception arises in storing the tables in hdf5 format , the assertion tests were just to show that no changes are made to the table which the user is using .

Reply:
> the assertion tests were just to show that no changes are made to the table which the user is using
Right, this is good then :)